dictionary.go

```diff
@@ -104,135 +51,56 @@ type statusLoginParam struct {
 }
 
 type statusSignupParam struct {
 	Password string `json:"password"`
 }
 
 type signupResponse struct {
-	Result struct {
-		Address  string `json:"address"`
-		Pubkey   string `json:"pubkey"`
-		Mnemonic string `json:"mnemonic"`
-	} `json:"result"`
+	Address  string `json:"address"`
+	Pubkey   string `json:"pubkey"`
+	Mnemonic string `json:"mnemonic"`
 }
 
-func statusSignupRequest(sdk *SDK, password string) (res signupResponse, err error) {
-	// `{"jsonrpc":"2.0","method":"status_signup","params":[{"password":"%s"}]}`
-	req := shhRequest{
-		JSONRPC: "2.0",
-		Method:  "status_signup",
-	}
+func statusSignupRequest(sdk *SDK, password string) (*signupResponse, error) {
+	var res signupResponse
 
-	req.Params = append(req.Params, &statusSignupParam{
+	params := &statusSignupParam{
 		Password: password,
-	})
-
-	body, err := json.Marshal(req)
-	if err != nil {
-		return
 	}
 
-	sdk.call(string(body), &res)
-
-	return
-}
-
-type getFilterMessagesResponse struct {
-	Result interface{} `json:"result"`
+	return &res, sdk.call("status_signup", params, &res)
 }
 
-func shhGetFilterMessagesRequest(sdk *SDK, filters []string) (res *getFilterMessagesResponse, err error) {
-	// `{"jsonrpc":"2.0","id":2968,"method":"shh_getFilterMessages","params":["%s"]}`
-	req := shhRequest{
-		ID:      2968,
-		JSONRPC: "2.0",
-		Method:  "shh_getFilterMessages",
-	}
-	for _, f := range filters {
-		req.Params = append(req.Params, f)
-	}
-
-	body, err := json.Marshal(req)
-	if err != nil {
-		return
-	}
-
-	sdk.call(string(body), &res)
-
-	return
+func shhGetFilterMessagesRequest(sdk *SDK, filters string) (interface{}, error) {
+	var res interface{}
+	return res, sdk.call("shh_getFilterMessages", filters, res)
 }
 
-type shhPostParam struct {
+type Message struct {
 	Signature string  `json:"sig"`
 	SymKeyID  string  `json:"symKeyID"`
 	Payload   string  `json:"payload"`
 	Topic     string  `json:"topic"`
-	TTL       int     `json:"ttl"`
+	TTL       uint32  `json:"ttl"`
 	PowTarget float64 `json:"powTarget"`
-	PowTime   int     `json:"powTime"`
-}
-
-// error response {"jsonrpc":"2.0","id":633,"error":{"code":-32000,"message":"message rejected, PoW too low"}}
-type sshPostError struct {
-	Code    float64 `json:"code"`
-	Message string  `json:"message"`
+	PowTime   uint32  `json:"powTime"`
 }
 
-type shhPostResponse struct {
-	Error *sshPostError `json:"error"`
-}
-
-func shhPostRequest(sdk *SDK, params []*shhPostParam) (res *shhPostResponse, err error) {
-	// `{"jsonrpc":"2.0","id":633,"method":"shh_post","params":[{"sig":"%s","symKeyID":"%s","payload":"%s","topic":"%s","ttl":10,"powTarget":%g,"powTime":1}]}`
-	req := shhRequest{
-		ID:      633,
-		JSONRPC: "2.0",
-		Method:  "shh_post",
-	}
-	for _, p := range params {
-		req.Params = append(req.Params, p)
-	}
-
-	body, err := json.Marshal(req)
-	if err != nil {
-		return
-	}
-
-	err = sdk.call(string(body), &res)
-	if err != nil {
-		return res, err
-	}
-
-	if res.Error != nil {
```

Comment:
How / where is this error being managed?

Reply:
@adriacidre the `RPCClient` interface defines `Call` as: `Call(req *Request, res interface{}) error`, so in case of errors, the error is returned and `res` is not used

Reply:
This error was checking a success response like `{"error":{"code":5000,"message":"oops, something broken on the endpoint"}}` and not an error on the call itself.

You can easily check this behaviour by trying to send a standard message with a nil  		`PowTarget` that will return something like `{"error":{"code":5000,"message":"Pow is too low"}}`